src/main/kotlin/io/github/mojira/arisa/ArisaMain.kt

```diff
@@ -185,29 +208,30 @@ private fun runIfWhitelisted(issue: Issue, projects: String, body: () -> Either<
     }
 
 private fun log0AndReturnUnit(method: String) = ({ Unit.right() }).also { log.info("[SHADOW] $method ran") }
-private fun log1AndReturnUnit(method: String) = { _: Any -> Unit.right() }.also { log.info("[SHADOW] $method ran") }
-private fun log2AndReturnUnit(method: String) = { _: Any, _: Any -> Unit.right() }.also { log.info("[SHADOW] $method ran") }
-private fun <T : () -> Either<Throwable, Unit>> runIfShadow(isShadow: Boolean, method: String, func: T) =
+private fun <T> log1AndReturnUnit(method: String) = { _: T -> Unit.right() }.also { log.info("[SHADOW] $method ran") }
+private fun <T, U> log2AndReturnUnit(method: String) = { _: T, _: U -> Unit.right() }.also { log.info("[SHADOW] $method ran") }
+
+private fun run0IfShadow(isShadow: Boolean, method: String, func: () -> Either<Throwable, Unit>) =
     if (!isShadow) {
         func
     } else {
         log0AndReturnUnit(method)
     }
 
-private fun <T : (Attachment) -> Either<Throwable, Unit>> runIfShadowAttachment(
+private fun <T> run1IfShadow(
```

Comment:
Hopefully we launch it soon and remove all of this shadow stuff, its pretty ugly to maintain haha